### PR TITLE
Feat bluetooth for friends (frontend part)

### PR DESCRIPTION
**Pull request by tercierp**
### Overview
Second bluetooth PR I focused mainly on the design of AddFriend to match the defini design on our Figma.
To see the more technical part on bluetooth management, go to [the first PR](https://github.com/SwEnt-Group8/Street-work-app/pull/103).
### Demo
Here's the `Dialog` pop-up when a user receives a friend request

<p align="center">
 image
</p>
Here's the AddFriend interface, which lets you add friends by activating the client and bluetooth server.

<p align="center">
 image
</p>

### CODE REVIEW

**Review by SaturneV — CHANGES_REQUESTED** [review]
Great PR! The UI is well done and adds an important feature to the app, thanks for taking care of it. I have two comments that I'll let you take into account, these being non-critical and are just details I'll let you choose what to do with them. I'll approve this PR as soon as you've responded to my comments. Good job!

File: app/src/main/java/com/android/streetworkapp/ui/profile/BluetoothButton.kt [hunk 1]
```diff
@@ -22,9 +26,10 @@ fun BluetoothButton(
   Button(
       onClick = {
         bluetoothController.startGattServer(uid) // Start GATT server to send UID
-        Toast.makeText(context, "Broadcasting UID via BLE GATT server", Toast.LENGTH_SHORT).show()
+        Toast.makeText(context, "Send friend request", Toast.LENGTH_SHORT).show()
```
[the remark this answers is not in the document]
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/ui/profile/BluetoothButton.kt:29]: I changed the message with this one "_Friend request sent_"

File: app/src/main/java/com/android/streetworkapp/ui/profile/AddFriend.kt [hunk 2]
```diff
@@ -27,46 +27,50 @@ import androidx.compose.runtime.remember
...
   val currentUser = userViewModel.currentUser.collectAsState().value
+  val user = userViewModel.user.collectAsState().value
   val uid = currentUser?.uid ?: ""
 
   // Instantiate BluetoothClient
   val bluetoothClient = remember { BluetoothClient(context) }
   val bluetoothServer = remember { BluetoothServer(context) }
 
+  var showRequestDialog by remember { mutableStateOf(false) }
+  var receivedUid2 by remember { mutableStateOf("") }
```
[the remark this answers is not in the document]
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/ui/profile/AddFriend.kt:65]: Yes, good catch! That was part of the debugging process, and I forgot to remove it afterward.

**Review by SaturneV — APPROVED** [review]
Thanks for responding to my comments and making the changes, it's good on my side. I approve it. Good job!

**Pull request merged**